### PR TITLE
Makefile rework + fix guidelines enforcer errors

### PR DESCRIPTION
**Pull request by apaillier-ledger**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -93,24 +88,16 @@ endif
...
-                DEFINES   += HAVE_PRINTF PRINTF=semihosted_printf
-        else
-                ifeq ($(TARGET_NAME),TARGET_NANOS)
-                        DEFINES   += HAVE_PRINTF PRINTF=screen_printf
-                else
-                        DEFINES   += HAVE_PRINTF PRINTF=mcu_usb_printf
-                endif
-
-        endif
+	DEFINES += HAVE_STACK_OVERFLOW_CHECK
```
Comment (xchapron-ledger) [inline Makefile]: Do you known where this is used?

File: Makefile [hunk 2]
```diff
@@ -93,24 +88,16 @@ endif
...
 # Enabling debug PRINTF
 ifneq ($(DEBUG),0)
-        DEFINES += HAVE_STACK_OVERFLOW_CHECK
-        SDK_SOURCE_PATH  += lib_stusb lib_stusb_impl
-        DEFINES   += HAVE_IO_USB HAVE_L4_USBLIB IO_USB_MAX_ENDPOINTS=4 IO_HID_EP_LENGTH=64 HAVE_USB_APDU
-
-        ifeq ($(DEBUG),10)
-                $(warning Using semihosted PRINTF. Only run with speculos!)
-                CFLAGS    += -include src/dbg/debug.h
-                DEFINES   += HAVE_PRINTF PRINTF=semihosted_printf
```
Comment (xchapron-ledger) [inline Makefile:103]: This is not used?

File: Makefile (hunk 1 above)
Comment (apaillier-ledger) [reply Makefile]: Indeed, it does not seem to be used by the plugin or the SDK.

File: Makefile (hunk 2 above)
Comment (apaillier-ledger) [reply Makefile:103]: Probably comes from a time where Speculos required this to have PRINTFs working.
It is not required anymore as the regular `mcu_usb_printf` works just fine.

File: Makefile [hunk 3]
```diff
@@ -1,173 +1,112 @@
...
+#ENABLE_NBGL_KEYPAD = 1
+
+########################################
+#          Features disablers          #
+########################################
+# These advanced settings allow to disable some feature that are by
+# default enabled in the SDK `Makefile.standard_app`.
+DISABLE_STANDARD_APP_FILES = 1
+#DISABLE_DEFAULT_IO_SEPROXY_BUFFER_SIZE = 1 # To allow custom size declaration
+DISABLE_STANDARD_APP_DEFINES = 1 # Will set all the following disablers
```
Comment (xchapron-ledger) [inline Makefile]: need to be changed to add a newline as you told me :)

File: Makefile [hunk 4]
```diff
@@ -1,173 +1,112 @@
...
-DEFINES   += HAVE_BAGL_FONT_OPEN_SANS_LIGHT_16PX
-DEFINES   += HAVE_UX_FLOW
-endif
-endif
-
-# Enabling debug PRINTF
-ifneq ($(DEBUG),0)
-        DEFINES += HAVE_STACK_OVERFLOW_CHECK
-        SDK_SOURCE_PATH  += lib_stusb lib_stusb_impl
-        DEFINES   += HAVE_IO_USB HAVE_L4_USBLIB IO_USB_MAX_ENDPOINTS=4 IO_HID_EP_LENGTH=64 HAVE_USB_APDU
```
Comment (xchapron-ledger) [inline Makefile:98]: Not needed anymoire for printf on speculos correct?

**Review by xchapron-ledger — COMMENTED** [review]
How much does it change the generated binary?

**Comment by apaillier-ledger** [comment]
> How much does it change the generated binary?

No change in the checksum of the resulting bin/app.hex

**Review by xchapron-ledger — DISMISSED** [review]

**Comment by xchapron-ledger** [comment]
> > How much does it change the generated binary?
> 
> No change in the checksum of the resulting bin/app.hex

Then what are we waiting for a merge!

**Pull request merged**